orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/WaitForTerminatedInstancesTask.groovy

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2014 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.tasks
+
+import com.netflix.spinnaker.orca.DefaultTaskResult
+import com.netflix.spinnaker.orca.ExecutionStatus
+import com.netflix.spinnaker.orca.RetryableTask
+import com.netflix.spinnaker.orca.TaskResult
+import com.netflix.spinnaker.orca.clouddriver.OortService
+import com.netflix.spinnaker.orca.pipeline.model.Stage
+import groovy.util.logging.Slf4j
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+import retrofit.RetrofitError
+
+@Slf4j
+@Component
+class WaitForTerminatedInstancesTask extends AbstractCloudProviderAwareTask implements RetryableTask {
+  long backoffPeriod = 1000
+  long timeout = 3600000
+
+  @Autowired
+  OortService oortService
+
+  @Override
+  TaskResult execute(Stage stage) {
+    def instanceIds = stage.context."terminate.instance.ids" as List<String>
+    def launchTimes = stage.context.launchTimes as List<Long>
+    if (!instanceIds || !launchTimes || instanceIds.size() != launchTimes.size()) {
+      return new DefaultTaskResult(ExecutionStatus.FAILED)
+    }
```

Comment:
Done. This involved a small conversion in the two orca terminate instance stages. Annoyingly, the two respective ops in clouddriver do not have the same APIs, therefor the conversion was slightly different for each. :: shrug ::